packages/react-core/src/demos/examples/TextInputGroup/AutoCompleteSearch.tsx

```diff
@@ -64,9 +65,24 @@ export const AutoCompleteSearch: React.FunctionComponent = () => {
         </MenuItem>
       );
       setMenuItems([noResultItem]);
+      setHint('');
       return;
     }
 
+    /** The hint is set whenever there is only one autocomplete option left. */
+    if (filteredMenuItems.length === 1) {
+      const hint = filteredMenuItems[0].props.children;
+      if (hint.indexOf(inputValue)) {
+        // the match was found in a place other than the start, so typeahead wouldn't work right
+        setHint('');
+      } else {
+        // use the input for the first part, otherwise case difference could make things look wrong
+        setHint(inputValue + hint.substr(inputValue.length));
```

Comment:
@thatblindgeye - I would expect "name" not to provide ghosting because there would be 2 autocomplete matches at that point (name and namespace), it would be getting stopped at the `filteredMenuItems.length === 1` condition in this case.

If you type "names" you should get the single match and the ghosting. As for the `indexaOf` logic, `hint.indexOf(inputValue)` would be non-zero (truthy) if the match is not at the beginning (which would make ghosting possible).

Reply:
Ah thanks for catching that. I had meant to say when I type "names". You're right, though, I think I was basing my suggested changes off of a combo of your original `indexOf` line and the updated one so thank you for pointing that out. 

After testing it some more it looks like I'm not getting the ghosting effect due to the difference in case ("name" vs "Name"), as when I type "Names" I get the ghosting to appear again. I have a screenshot showing the two side by side:

![image](https://user-images.githubusercontent.com/70952936/151998085-4cf9c2c2-cabe-4ac3-92c0-39cf2d05f362.png)

If that's the expected behavior I can mark this as resolved, though.